### ISSUE TITLE
Missing File Transfer Packets

### ISSUE THREAD

**Issue by brandon1024** [brandon1024/Java-IRC#6]
When sending a large image, it looks as though some of the packets may have had been dropped or lost. This was tested by sending an image (13.9 MB, 8518 x 5679) and comparing the image before and after the transfer. Implementing a corrupted transfer checker is also a good idea.

Before Transfer:
![earth-clouds2700](https://cloud.githubusercontent.com/assets/22732449/23823203/06e5e38a-0634-11e7-938a-8c74a36eda28.jpg)

After Transfer:
![earth-clouds2700](https://cloud.githubusercontent.com/assets/22732449/23823208/1e11be4e-0634-11e7-94c7-ffa98c453d91.jpg)

Caused by: [Widespread Code Refactor](https://github.com/brandon1024/Java-IRC/pull/2)

**Comment by brandon1024** [comment]
Corrupted Image Stacktrace:
```
sun.awt.image.PNGImageDecoder$PNGException: crc corruption
	at sun.awt.image.PNGImageDecoder.getChunk(PNGImageDecoder.java:717)
	at sun.awt.image.PNGImageDecoder.getData(PNGImageDecoder.java:725)
	at sun.awt.image.PNGImageDecoder.produceImage(PNGImageDecoder.java:252)
	at sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:269)
	at sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:205)
	at sun.awt.image.ImageFetcher.run(ImageFetcher.java:169)
sun.awt.image.PNGImageDecoder$PNGException: crc corruption
	at sun.awt.image.PNGImageDecoder.getChunk(PNGImageDecoder.java:717)
	at sun.awt.image.PNGImageDecoder.getData(PNGImageDecoder.java:725)
	at sun.awt.image.PNGImageDecoder.produceImage(PNGImageDecoder.java:252)
	at sun.awt.image.InputStreamImageSource.doFetch(InputStreamImageSource.java:269)
	at sun.awt.image.ImageFetcher.fetchloop(ImageFetcher.java:205)
	at sun.awt.image.ImageFetcher.run(ImageFetcher.java:169)
```

**Comment by brandon1024** [comment]
By adding a packet number field to the `TransferBuffer` class, I found that buffers are not getting stored in the queue in the correct order. The order with which the buffers are stored is irregular, changing with each transfer. Below is one example of a series of mismatched buffers:
```
PREV: 1627     CURR: 1626
PREV: 1626     CURR: 1628
PREV: 4436     CURR: 4438
PREV: 6640     CURR: 6639
PREV: 6639     CURR: 6641
```

The final buffer processed by the `InboundFileTranfserExecutor` has a length of 4096, while the final buffer sent by the `OutboundFileTransferExecutor` was 2790, which further indicates that the order of queued buffers is incorrect. This is caused by concurrency issues with DynamicQueue in intensive workloads. Synchronizing the queue methods solves this issue.

**Comment by brandon1024** [comment]
Fixed: [#6 Missing File Transfer Packets](https://github.com/brandon1024/Java-IRC/pull/17)
